### PR TITLE
Jeremy Duncan week-2-assessment

### PR DESCRIPTION
**Pull request by JeremyDuncan**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [✅] Coding Challenges
- [✅] Coding Concepts
- [✅] Interview Questions

### 📝 Feedback 🤘

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -2,58 +2,232 @@
...
+  it("returns the negative object number is divisable by three..", () => {
+    expect(divByThree(object4)).toEqual("-9 is divisible by three");
+  });
+  it("returns positive object number is not divisable by three..", () => {
+    expect(divByThree(object5)).toEqual("10 is not divisible by three");
+  });
+});
+
+// -----------------=== Provided Test Cases ===---------------------------------
+const object1 = { number: 15 };
```
Comment (elysemontano) [inline code-challenges.test.js:62]: I like that you are adding edge cases and tests to account for them!  Since these test variables are only being utilized within your test, it is best practice to place the variables in your test so they are only accessible within the scope of the test.

File: code-challenges.test.js [hunk 2]
```diff
@@ -2,58 +2,232 @@
...
 
+// =============================================================================
+// b) Create the function that makes the test pass.
+// =============================================================================
+
+//---------------------== Function Explanation ==-------------------------------
+// -----Inputs: Object
+// -----Output: String
+//
+// I decided to be more concise and use a ternary operator for this problem
```
Comment (elysemontano) [inline code-challenges.test.js:83]: Awesome work in using the ternary operator!

File: code-challenges.test.js [hunk 3]
```diff
@@ -2,58 +2,232 @@
...
+// Expected output: null
+const vowelTester5 = "LEARN";
+
+// Expected output: null
+// =============================================================================
 // b) Create the function that makes the test pass.
+// =============================================================================
+
+//---------------------== Function Explanation ==-------------------------------
+// Input: String
```
Comment (elysemontano) [inline code-challenges.test.js:203]: Excellent work in this section!  Your pseudocode is excellent and very easy to read, and your tests, code and logic look great!  If you want to stretch yourself, consider using a higher-order function for this last problem.

File: coding-concepts.js [hunk 4]
```diff
@@ -3,66 +3,233 @@
...
+// onlyOdds, and so forth with 13, and 15 inside a new array.
+//
 // b) Verify and explain:
-
+// according to: MDN Webdocs, filter() creates a shallow copy af an array based
+// elements that pass whatever test may be implemenmted by a function inside the
+// filter() method.
+//
+// *** Rabbit Hole ***
+// This research led me down a rabbit hole as I was curious about what a
```
Comment (elysemontano) [inline coding-concepts.js:137]: Super interesting, thanks for sharing!

File: coding-concepts.js [hunk 5]
```diff
@@ -3,66 +3,233 @@
...
+// and object with your own declared values by passing on arguments.
+//
+// const learnStudentWithArguments = new Learn("Sally", "FoxWho?", 1999);
+// Object RESULT => Learn { student: "Sally", cohort: "FoxWho?", year: 1999 }
+//
+// const learnStudentDefault = new Learn();
+// Object RESULT => Learn { student: "Billy", cohort: "Foxtrot", year: 2022 }
+
 // b) Verify and explain:
+// according to: MDN Webdocs, classes are "special functions" I found that
```
Comment (elysemontano) [inline coding-concepts.js:227]: I really appreciate all of the thorough research that you put into these questions!  Excellent work!

File: interview-questions.md [hunk 6]
```diff
@@ -2,56 +2,233 @@
...
+---
+<!-- ----------------------------------------------------------------------- -->
+## 3. What is the difference between map and filter?
+
+### Your answer: 
+* map() affects every element in an array
+  * You would use map() to invite everyone in your office to your house party..
+
+* filter() effects only the elements that pass a given conditional test
+  * You would use filter() to invite everyone but Dwight..
```
Comment (elysemontano) [inline interview-questions.md:85]: 🤣 Love it!!

**Review by elysemontano — APPROVED** [review]
Crushed this assessment Jeremy!  Such thorough research and pseudocode and your code is clean and efficient.  I really appreciate the detail you put into each problem, and of course the added humor is also appreciated as well!  Great work!
